src/lib/y2network/connection_config/wireless.rb

```diff
@@ -0,0 +1,63 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/connection_config/base"
+
+module Y2Network
+  module ConnectionConfig
+    # Configuration for wireless connections
+    class Wireless < Base
+      # wireless options
+      # @return [String]
+      attr_accessor :mode
```

Comment:
maybe enum instead of string? at least add fixme.

Reply:
I will add the FIXME, in general I just went from LanItems variables moving here the wireless specific, but not going too much into details.

So when we start to add some real behavior, or specific use of this class we will be able to rethink the current internal state.